src/libraries/System.Net.Quic/ref/System.Net.Quic.cs

```diff
@@ -70,4 +78,19 @@ public class QuicListenerOptions
         public long MaxUnidirectionalStreams { get => throw null; set => throw null; }
         public TimeSpan IdleTimeout { get => throw null; set => throw null; }
     }
+    public class QuicException : Exception
+    {
+        public QuicException(string message) : base(message) { }
+    }
+    public class QuicConnectionAbortedException : QuicException
```

Comment:
There are two CONNECTION_CLOSE frame types indicating either a quic-level or application-level error. I'm not sure what distinguishes these yet but we should figure out if we need to indicate that here.

Reply:
Yeah, I still wasn't certain about how to propagate the QuicConnectionAbortedException yet. Do you think the application would care if it was aborted by the application itself versus the transport?

Reply:
We may want a `Task<long> Completed` on `QuicConnection` so that server can observe a connection close when you have no active streams.

Reply:
Actually I guess you'd get an exception via `AcceptStream()`, nevermind.

Reply:
> Do you think the application would care if it was aborted by the application itself versus the transport?

I don't believe HTTP/3 makes a distinction, so we can try to get by without it for now.

Reply:
Kk. I'll make AcceptStream and OpenBi/UniStream throw QuicConnectionAbortedException then.

Reply:
Do we care about making the distinction between graceful vs abortive connection closure? Ex: if someone calls CloseAsync, if they have a pending call to AcceptStream, do you want that to throw an QuicConnectionAbortedException?

Reply:
I can see it being useful to distinguish between a reset that was raised locally vs remotely.